### ISSUE TITLE
Linking error when inserting use rp2040_flash::flash

### ISSUE THREAD

**Issue by papyDoctor** [jannic/rp2040-flash#5]
I'm using the crate with rtic (cortex-m-rtic = { version = "1.1.2" })
and I've a linking error just by inserting in the code
`use rp2040_flash::flash;`
Linking error:
`warning: Linking globals named '_critical_section_acquire': symbol multiply defined!
error: failed to load bitcode of module "rp2040_hal-f8e9e70b849379ad.rp2040_hal.dba293a0-cgu.0.rcgu.o": `
Do you have any suggestion about this?

**Comment by jannic** [comment]
Yes, this is because rp2040-flash depends on rp2040-hal 0.6.0, and that version doesn't yet include https://github.com/rp-rs/rp-hal/pull/444, so it defines `_critical_section_acquire` by default.

I'll change rp2040-flash to depend on rp2040-hal with `default-features = false`.

**Comment by jannic** [comment]
Released version 0.1.1 with that change.

**Comment by papyDoctor** [comment]
Just for someone that wants to store data at the end of the flash memory (to be sure that at each firmware update, the address data is the same and then the data still can be read), modify the memory.x linker script like this:
```
MEMORY {
    BOOT2 : ORIGIN = 0x10000000, LENGTH = 0x100
    FLASH : ORIGIN = 0x10000100, LENGTH = 2048K - 0x100
    FLASHIAP : ORIGIN = 0x101FF000, LENGTH = 4K
    RAM   : ORIGIN = 0x20000000, LENGTH = 256K
}
EXTERN(BOOT2_FIRMWARE)
SECTIONS {
    /* ### Boot loader */
    .boot2 ORIGIN(BOOT2) :
    {
        KEEP(*(.boot2));
    } > BOOT2
} INSERT BEFORE .text;
SECTIONS {
    /* ### Flash IAP */
    .flashIAP ORIGIN(FLASHIAP) :
    {
        KEEP(*(.flashIAP));
    } > FLASHIAP
}
```
and in the code (see example here), declare the var like this:
```
#[link_section = ".flashIAP"]
    static TEST: FlashBlock = FlashBlock {
        data: UnsafeCell::new([0x55u8; 4096]),
    };
```
Run example code, you'll see:
`INFO  Addr of flash block is 101ff000`

I hope I didn't made any mistake
